cmake/googletest.cmake

```diff
@@ -0,0 +1,38 @@
+# SPDX-License-Identifier: BSD-3-Clause "New" or "Revised" License
+#
+# SPDX-FileCopyrightText: 2008, Google Inc. (see AUTHORS.md)
+
+# GoogleTest
+# https://github.com/google/googletest/blob/master/googletest/README.md#incorporating-into-an-existing-cmake-project
+
+# Download and unpack googletest at configure time
```

Comment:
Why do we need download the framework as part of the configuration stage and not just have the `ExternalProject_Add` in a regular cmake?

Reply:
We've initially used the approach recommended by Google, but probably we'll have to think this again because of several reasons, among them: 
- downloading GoogleTest framework on each configuration is slower than having a pre-downloaded copy.
- cppcheck is reporting errors when analyzing unit tests  because it doesn't have the sources for the Google Test framework (temporarily patched by ignoring unit tests). 

We've decided to merge this PR as is so others can start using GoogleTest ASAP and then improve this issue and some others that have arisen during this review.